staging/src/k8s.io/component-base/logs/config.go

```diff
@@ -40,7 +39,6 @@ var LogRegistry = NewLogFormatRegistry()
 func init() {
 	// Text format is default klog format
 	LogRegistry.Register(DefaultLogFormat, nil)
-	LogRegistry.Register(JSONLogFormat, json.JSONLogger)
```

Comment:
once json logging graduates to stable, won't we want to add this back in? from the discussion in https://github.com/kubernetes/kubernetes/pull/102869/files#r651946267, it sounds like the package structure has changed in a way that will make that difficult

Reply:
If it becomes a GA, it also requires that each component be explicitly registered.
Or does this row need to be added back? 

Any better suggestions? @serathius

Reply:
I think it's better to set a precedence that alternative logging formats are optional for component owner to decide to support. As they grow in number, golang dependencies will grow in number, keeping all formats even GAed installed by default will not scale. We have already forced all users of component-base to depend on zap. 

I think long term we should go into direction of having separate packages for logging fomats each with minimal dependencies, and allow components owners which ones they want to maintain.

Reply:
> I think long term we should go into direction of having separate packages for logging fomats each with minimal dependencies, and allow components owners which ones they want to maintain.

ok... in that case, the tests should probably move to that package as well instead of doing test package trickery from the central logs package

Reply:
That's a good point. We should move tests that validate configuring flag to json/register package. This way the we remain tests for flag validation, and register has it's own tests.

Reply:
had moved tests that validate configuring flag to json/register package.
log/options_test.go only tests the default text flag and to avoid the need to import JSON/Register packages.
Is that the right thing to do？ Do I understand you correctly？@serathius

Reply:
SG